src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2091,42 +2091,47 @@ def process_declarations(self, parent, nodes, arg_list,
-                    # being put into a CodeBlock.
-                    raise NotImplementedError()
-                name = node.children[0].children[0].string.lower()
-                vis = visibility_map.get(
-                    name, parent.symbol_table.default_visibility)
-                # A named interface block corresponds to a RoutineSymbol.
-                # (There will be calls to it although there will be no
-                # corresponding implementation with that name.)
-                # We store its definition using an UnknownFortranType so that
-                # we can recreate it in the Fortran backend.
-                try:
-                    parent.symbol_table.add(
-                        RoutineSymbol(
-                            name, UnknownFortranType(str(node).lower()),
-                            visibility=vis))
-                except KeyError:
-                    # This symbol has already been declared. However
-                    # we still want to output the interface so we
-                    # store it in the PSyIR as an UnkownFortranType
-                    # with an internal name.
+                if not isinstance(node.children[0].children[0],
+                                  Fortran2003.Name):
+                    # This interface has no name but we still want to
+                    # store it in the PSyIR. Therefore we store its
+                    # definition in an UnknownFortranType with an
+                    # internal name so we can recreate it in the
+                    # Fortran backend.
                     parent.symbol_table.new_symbol(
-                        root_name=f"_psyclone_internal_{name}",
+                        root_name="_psyclone_internal_interface",
                         symbol_type=RoutineSymbol,
-                        datatype=UnknownFortranType(str(node).lower()),
-                        visibility=vis)
+                        datatype=UnknownFortranType(str(node).lower()))
+                else:
+                    # This interface has a name.
+                    name = node.children[0].children[0].string.lower()
+                    vis = visibility_map.get(
+                        name, parent.symbol_table.default_visibility)
+                    # A named interface block corresponds to a
+                    # RoutineSymbol. (There will be calls to it
+                    # although there will be no corresponding
+                    # implementation with that name.) We store its
+                    # definition using an UnknownFortranType.
+                    try:
+                        parent.symbol_table.add(
+                            RoutineSymbol(
+                                name, UnknownFortranType(str(node).lower()),
+                                visibility=vis))
+                    except KeyError:
+                        # This symbol has already been declared. However
```

Comment:
I know I was the one who wrote this originally but I can't remember now: do you know why it might already have been declared? If so, please could you extend the comment.

Reply:
Actually I wrote this so it's not your fault. I've added some words. The case that we need to support is when the interface overloads a routine within a type. In this case the interface name is the name of the type.